### PR TITLE
Correct the SyntaxError on old navigators

### LINKED ISSUE #23743: Keycloak  22.0.4 crashes on file step1.html on old safari versions
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

authentication

### Describe the bug

On old safari (safari 14 and older), the file 3p-cookies-step1.html crashes because there's an await at the root of the script.

Here is the console error:

SyntaxError: Unexpected identifier 'hasStorageAccess'. Expected ';' after variable declaration.

### Version

22.0.4

### Expected behavior

Keycloak runs on old navigators

### Actual behavior

Keycloak crashes on old navigators. Users see a white page

### How to Reproduce?

try to connect to a client on a safari older than safari 15

### Anything else?

I thin this could fix it:

`<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
  </head>
  <body>
    <script type="module">
      attemptCookie();

     // Encapsulate the code in an async to avoid syntax error
      async function attemptCookie() {
        // Check if the browser has granted us access to 3rd-party storage (such as cookies).
        const hasAccess = await hasStorageAccess();

        if (hasAccess) {
          // If so, attempt to place a cookie to test this assumption.
          attemptWithTestCookie();
        } else {
          // Otherwise, signal that 3rd-party access is not supported.
          signalSupport(false);
        }
      }

      async function hasStorageAccess() {
        if (!("hasStorageAccess" in document)) {
          return true;
        }

        return document.hasStorageAccess();
      }

      function attemptWithTestCookie() {
        // Place a cookie to test whether we can access cookies from 3rd-party storage.
        document.cookie =
          "KEYCLOAK_3P_COOKIE_SAMESITE=supported; Max-Age=60; SameSite=None; Secure";
        document.cookie = "KEYCLOAK_3P_COOKIE=supported; Max-Age=60";

        // Then redirect to the page where we will read these cookies to confirm this.
        window.location = "step2.html";
      }

      function signalSupport(isSupported) {
        // Signal 3rd-party access support to the page embedding this iframe.
        window.parent.postMessage(
          isSupported ? "supported" : "unsupported",
          "*"
        );
      }
    </script>
  </body>
</html>
`

Can you give me the protocol to make a valid PR?

### PR DESCRIPTION
**Pull request by ici-dev-gb**
Closes #23743

### CODE REVIEW

File: services/src/main/resources/org/keycloak/protocol/oidc/endpoints/3p-cookies-step1.html [hunk 1]
```diff
@@ -5,15 +5,20 @@
   </head>
   <body>
     <script type="module">
-      // Check if the browser has granted us access to 3rd-party storage (such as cookies).
-      const hasAccess = await hasStorageAccess();
+      attemptCookie();
+      
+     // Encapsulate the code in an async to avoid syntax error
```
Comment (jonkoops) [inline services/src/main/resources/org/keycloak/protocol/oidc/endpoints/3p-cookies-step1.html:10]: I don't think we need a comment for this, this is also exclusive to older versions of certain browsers.

```suggestion
```

File: services/src/main/resources/org/keycloak/protocol/oidc/endpoints/3p-cookies-step1.html [hunk 2]
```diff
@@ -5,15 +5,20 @@
...
+      async function attemptCookie() {
```
Comment (jonkoops) [inline services/src/main/resources/org/keycloak/protocol/oidc/endpoints/3p-cookies-step1.html:11]: I'd rather call this `checkStorageAccess()`, as the cookies are an implementation detail.
```suggestion
      async function checkStorageAccess() {
```

File: services/src/main/resources/org/keycloak/protocol/oidc/endpoints/3p-cookies-step1.html [hunk 3]
```diff
@@ -35,7 +40,7 @@
 
       function signalSupport(isSupported) {
         // Signal 3rd-party access support to the page embedding this iframe.
-        window.parent.postMessage(isSupported ? "supported" : "unsupported", "*");
+        window.parent.postMessage( isSupported ? "supported" : "unsupported", "*");
```
Comment (jonkoops) [inline services/src/main/resources/org/keycloak/protocol/oidc/endpoints/3p-cookies-step1.html:43]: Can we get rid of these white-space changes?
```suggestion
        window.parent.postMessage(isSupported ? "supported" : "unsupported", "*");
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Comment by jonkoops** [comment]
Normally we would not guarantee support for more than the last two major versions of Safari. But considering this a very mission critical path for clients and the fix doesn't compromise the code quality, I think we can make an exception here.

File: services/src/main/resources/org/keycloak/protocol/oidc/endpoints/3p-cookies-step1.html [hunk 4]
```diff
@@ -5,15 +5,19 @@
   </head>
   <body>
     <script type="module">
-      // Check if the browser has granted us access to 3rd-party storage (such as cookies).
-      const hasAccess = await hasStorageAccess();
+      attemptCookie();
```
Comment (jonkoops) [inline services/src/main/resources/org/keycloak/protocol/oidc/endpoints/3p-cookies-step1.html:8]: This function no longer exists, it should be calling `checkStorageAccess()` instead.
```suggestion
      checkStorageAccess();
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Review by jonkoops — APPROVED** [review]

**Review by ahus1 — APPROVED** [review]
Approving as of @jonkoops's review

**Pull request merged**